### PR TITLE
Nabalas/newdev

### PR DESCRIPTION
**Pull request by NarmathaBala**
This branch merges changes from main repo and consolidates changes from the team. This contains

1. Documentation updates - README and CONTRIBUTING
2. Group within and across columns
3. Grouped items in focus mode
4. Code coverage badge
5. Deploy alerts as IAC (Backend changes)
6. Add CD workflow for frontend
7. Unit test updates

### CODE REVIEW

File: .github/workflows/ci_frontend.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+      - if: steps.changes.outputs.src == 'true'
+        name: Get Coverage for badge
+        working-directory: 'RetrospectiveExtension.Frontend'
+        run: |
+          SUMMARY="$(npm test -- --coverageReporters='text-summary' | tail -2 | head -1)"
+          TOKENS=($SUMMARY)
+          # process.env.COVERAGE = '31.93%';
+          echo "COVERAGE=$(echo ${TOKENS[2]})" >> $GITHUB_ENV
+          # var REF = 'refs/pull/27/merge.json';
```
Comment (mindlessroman) [inline .github/workflows/ci_frontend.yml:62]: It seems like this is either lingering code or we should say in the comment it's an example.

File: .gitignore [hunk 2]
```diff
@@ -193,6 +193,7 @@ ClientBin/
 *.publishsettings
 node_modules/
 orleans.codegen.cs
+.DS_Store
```
Comment (mindlessroman) [inline .gitignore]: Don't need this twice.

File: RetrospectiveExtension.Frontend/dal/reflectBackendService.tsx [hunk 3]
```diff
@@ -60,31 +61,34 @@ class ReflectBackendService {
...
-      const tokenData = jsonwebtoken.decode(that._appToken.token, {json: true});
-      if (typeof tokenData === 'object') {
+      const tokenData = decodeJwt(that._appToken);
+      if (tokenData) {
         that._tokenExpiry = moment.unix(tokenData.exp).toDate();
-        return that._appToken.token;
+        return that._appToken;
       }
 
+      // TODO (phongcao) : appInsightsClient.trackException(new Error(e.message));
```
Comment (mindlessroman) [inline RetrospectiveExtension.Frontend/dal/reflectBackendService.tsx:91]: @phongcao  - just as a headsup this TODO got in

File: RetrospectiveExtension.Frontend/index.tsx [hunk 4]
```diff
@@ -1,23 +1,30 @@
-﻿/// <reference types="vss-web-extension-sdk" />
```
Comment (mindlessroman) [inline RetrospectiveExtension.Frontend/index.tsx:1]: Actually for the empty file `eslint` - can we remove?

**Review by mindlessroman — CHANGES_REQUESTED** [review]
Just a few things to resolve! I'M SO EXCITED

File: RetrospectiveExtension.Frontend/dal/reflectBackendService.tsx (hunk 3 above)
Comment (phongcao) [reply RetrospectiveExtension.Frontend/dal/reflectBackendService.tsx:91]: yeah it's there as a reminder for us to add AppInsights to this file in the future.

Comment (phongcao) [reply RetrospectiveExtension.Frontend/dal/reflectBackendService.tsx:91]: Resolved as will not fixed for now.

File: .github/workflows/ci_frontend.yml (hunk 1 above)
Comment (phongcao) [reply .github/workflows/ci_frontend.yml:62]: We haven't tested this coverage code on the main repo so something to keep in mind for checking.

**Review by phongcao — APPROVED** [review]

**Comment by phongcao** [comment]
@NarmathaBala Can you sync with @polatengin to switch GitHub Actions CI/CD to the new files?

**Review by mindlessroman — APPROVED** [review]

**Comment by mindlessroman** [comment]
Let's GOOOOOOO

**Pull request merged**